packages/react/src/components/Dropdown/utilities/DropdownSizePosCache.ts

```diff
@@ -62,14 +70,18 @@ export class DropdownSizePosCache {
     // we could possibly memoize this too but this should be good enough, most of the time (the expectation is that
     // when you have a lot of options, the selectable options will heavily dominate over the non-selectable options.
     let offset = 0;
-    while (index > this._displayOnlyOptionsCache[offset]) {
+    while (index > this._notSelectableOptionsCache[offset]) {
```

Comment:
Ignore all non-selectable options, not only Headers and Dividers but also `hidden` options.